packages/plugin/src/theia.d.ts

```diff
@@ -9691,7 +9698,11 @@ export module '@theia/plugin' {
          * @param token A cancellation token.
          * @return The resolved debug configuration or undefined or null.
          */
-        resolveDebugConfigurationWithSubstitutedVariables?(folder: WorkspaceFolder | undefined, debugConfiguration: DebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration>;
+        resolveDebugConfigurationWithSubstitutedVariables?(
+            folder: WorkspaceFolder | undefined,
+            debugConfiguration: DebugConfiguration,
+            token?: CancellationToken
+        ): ProviderResult<DebugConfiguration | undefined | null>;
```

Comment:
@alvsan09 I don't believe this api needs to be update since it is already aligned with [vscode](https://github.com/microsoft/vscode/blob/a3d53f76790d8c88f835afe85225ff6442703409/src/vscode-dts/vscode.d.ts#L13655-L13699) (and doing so will likely mark our api as incompatible in the compatibility report). Rather, the [`ProviderResult`](https://github.com/microsoft/vscode/blob/a3d53f76790d8c88f835afe85225ff6442703409/src/vscode-dts/vscode.d.ts#L2135) should be updated to accept `null` like in vscode:

```ts
export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>;
```

Reply:
Nice catch Vnce! Thanks!
It is addressed in today's commit.